include/drivers/spi.h

```diff
@@ -142,6 +142,80 @@ struct spi_cs_control {
 	gpio_dt_flags_t		gpio_dt_flags;
 };
 
+/**
+ * @brief Initialize and get a pointer to a @p spi_cs_control from a
+ *        devicetree node identifier
+ *
+ * This helper is useful for initializing a device on a SPI bus. It
+ * initializes a struct spi_cs_control and returns a pointer to it.
+ * Here, @p node_id is a node identifier for a SPI device, not a SPI
+ * controller.
+ *
+ * Example devicetree fragment:
+ *
+ *     spi@... {
+ *             cs-gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+ *             spidev: spi-device@0 { ... };
+ *     };
+ *
+ * Assume that @p gpio0 follows the standard convention for specifying
+ * GPIOs, i.e. it has the following in its binding:
+ *
+ *     gpio-cells:
+ *     - pin
+ *     - flags
+ *
+ * Example usage:
+ *
+ *     struct spi_cs_control *ctrl =
```

Comment:
I'm seeing only one use of this idiom in Zephyr, at https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/drivers/spi/spi_loopback/src/spi.c#L26.  Based on the concerns from https://github.com/zephyrproject-rtos/zephyr/pull/30536#discussion_r576809680 perhaps it would be better to make this macro support:

    .cs_ctrl = SPI_CS_CONTROL_DT(DT_NODELABEL(spidev), 2),

Then `&SPI_CS_CONTROL_DT(DT_NODELABEL(spidev), 2)` could be used in the apparently rare cases where  a pointer-to-object would be needed.

Reply:
> I'm seeing only one use of this idiom in Zephyr, at https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/drivers/spi/spi_loopback/src/spi.c#L26

Did you miss the one in SPI_CONFIG_DT, also added by this PR?

> .cs_ctrl = SPI_CS_CONTROL_DT(DT_NODELABEL(spidev), 2),

That could be done, but it would make the drivers uglier. I would need to make SPI_CONFIG_DT take a pointer to the spi_cs_control which is allocated manually elsewhere by the driver using SPI_CS_CONTROL_DT, instead of having SPI_CONFIG_DT just be a standalone macro which creates the entire spi_config.

Is C++ support really a blocker here?

Reply:
In my C++ code, I made a macro like this to get it running.
```C
SPI_CS_CONTROL_DT(node_id, delay_)					\
	{								\
	.gpio_dev = DEVICE_DT_GET(					\
		DT_SPI_DEV_CS_GPIOS_CTLR(node_id)),			\
	.delay = (delay_),						\
	.gpio_pin = DT_SPI_DEV_CS_GPIOS_PIN(node_id),			\
	.gpio_dt_flags = DT_SPI_DEV_CS_GPIOS_FLAGS(node_id),		\
}
```
and the use
```C++
static constexpr spi_cs_control cs_ctrl = SPI_CS_CONTROL_DT(DT_NODELLABEL(spidev), 0);
```
That works at least for C++20